Tests/MapboxMapsTests/Foundation/ViewAnnotationManagerTests.swift

```diff
@@ -0,0 +1,261 @@
+    }
+
+    // MARK: Test placeAnnotations
+
+    func testPlacementMissingAnnotation() {
+        manager.onViewAnnotationPositionsUpdate(forPositions: [ViewAnnotationPositionDescriptor(
+            identifier: "arbitraryId",
+            width: 0,
+            height: 0,
+            leftTopCoordinate: CGPoint(x: 0.0, y: 0.0)
+        )])
+    }
+
+    func testPlacementPosition() {
+        let annotationView = addTestAnnotationView()
+        XCTAssertEqual(container.subviews.count, 1)
+        XCTAssertEqual(annotationView.frame, CGRect.zero)
+
+        manager.onViewAnnotationPositionsUpdate(forPositions: [ViewAnnotationPositionDescriptor(
+            identifier: "0",
+            width: 100,
+            height: 50,
+            leftTopCoordinate: CGPoint(x: 150.0, y: 200.0)
+        )])
+
+        XCTAssertEqual(annotationView.frame, CGRect(x: 150.0, y: 200.0, width: 100.0, height: 50.0))
+    }
+
+    func testPlacementHideMissingAnnotations() {
+        let annotationViewA = addTestAnnotationView()
+        let annotationViewB = addTestAnnotationView()
+        let annotationViewC = addTestAnnotationView()
+
+        XCTAssertFalse(annotationViewA.isHidden)
+        XCTAssertFalse(annotationViewB.isHidden)
+        XCTAssertFalse(annotationViewC.isHidden)
+
+        manager.onViewAnnotationPositionsUpdate(forPositions: [ViewAnnotationPositionDescriptor(
+            identifier: "0",
+            width: 100,
+            height: 50,
+            leftTopCoordinate: CGPoint(x: 150.0, y: 200.0)
+        )])
+
+        XCTAssertFalse(annotationViewA.isHidden)
+        XCTAssertTrue(annotationViewB.isHidden)
+        XCTAssertTrue(annotationViewC.isHidden)
+    }
+
+    // MARK: - Helper functions
+
+    private func addTestAnnotationView(featureId: String? = nil) -> UIView {
+        let geometry = Geometry.point(Point(CLLocationCoordinate2D(latitude: 0.0, longitude: 0.0)))
+        let options = ViewAnnotationOptions(geometry: geometry, associatedFeatureId: featureId)
+        let view = UIView()
+        try! manager.add(view, options: options)
+        mapboxMap.optionsForViewAnnotationWithIdStub.defaultReturnValue = options
+        return view
+    }
+
+    private func triggerPositionUpdate(forId id: String) {
+        manager.onViewAnnotationPositionsUpdate(forPositions: [ViewAnnotationPositionDescriptor(
+            identifier: id,
+            width: 100,
+            height: 50,
+            leftTopCoordinate: CGPoint(x: 150.0, y: 200.0)
+        )])
+    }
+
+}
+
+fileprivate extension MapboxCoreMaps.ViewAnnotationPositionDescriptor {
+    convenience init(identifier: String, width: Int, height: Int, leftTopCoordinate: CGPoint) {
+        self.init(__identifier: identifier,
+                  width: UInt32(width),
+                  height: UInt32(height),
+                  leftTopCoordinate: ScreenCoordinate(x: leftTopCoordinate.x, y: leftTopCoordinate.y)
+        )
+    }
+}
```

Comment:
Let's move this to a separate file so it can be reused. We can put it in `/Tests/MapboxMapsTests/Helpers/Extensions/ViewAnnotationPositionDescriptor.swift`